python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -489,8 +489,19 @@ def get_sensor_def(self, name: str) -> SensorDefinition:
         If your passed-in sensor had resource dependencies, or the job targeted by the sensor had
         resource dependencies, those resource dependencies will be fully resolved on the returned object.
         """
+        warnings.warn(
+            "Starting in dagster 1.11, get_sensor_def will return a SensorDefinition without resolving resource dependencies on it or its target."
+        )
+
         return self.resolve_sensor_def(name)
 
+    # TODO: after dagster 1.11, this will become the implementation of get_sensor_def -- schrockn 2025-06-02
```

Comment:
if you don't already, recommend stacking the PR that does this cutover ahead of time and just sitting on it til the 1.11 branch cut to help prevent losing track

Reply:
What is the easiest way to do that? I've never done that process

Reply:
https://app.graphite.dev/github/pr/dagster-io/dagster/30456